src/cluster.c

```diff
@@ -1483,11 +1483,136 @@ static void addClusterSlotsReply(RedisRaftCtx *rr, RedisModuleCtx *ctx)
     RedisModule_ReplySetArrayLength(ctx, num_slots);
 }
 
+static void addClusterShardsNodeReply(RedisRaftCtx *rr, RedisModuleCtx *ctx, char *id, uint16_t port, char *host, char *role, int offset, char *health)
+{
+    RedisModule_ReplyWithMap(ctx, 8);
+    RedisModule_ReplyWithCString(ctx, "id");
+    RedisModule_ReplyWithCString(ctx, id);
+    if (!rr->config.tls_enabled) {
+        RedisModule_ReplyWithCString(ctx, "port");
+        RedisModule_ReplyWithLongLong(ctx, port);
+    } else {
+        RedisModule_ReplyWithCString(ctx, "tls-port");
+        RedisModule_ReplyWithLongLong(ctx, port);
+    }
+    RedisModule_ReplyWithCString(ctx, "ip");
+    RedisModule_ReplyWithCString(ctx, "");
+    RedisModule_ReplyWithCString(ctx, "hostname");
+    RedisModule_ReplyWithCString(ctx, host);
```

Comment:
IIRC we've decided to put our address in "ip" field as it is a mandatory field. 
I think we shouldn't return "hostname" field as our address can be IP. (this is not a mandatory field anyway).

In summary, we should reply with just these two fields:
1- "endpoint"
2- "ip"